src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/HttpScheme.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Http
+{
+    public enum HttpScheme
+    {
+        Unknown = -1,
+        Http = 0,
+        HttpSecure = 1
```

Comment:
How strong is your preference for this? I'm chose the name to make it super obvious that it's http + TLS, not plain-old http.